docs/contributing/docs-and-website-components.md

```diff
@@ -0,0 +1,418 @@
+
+#### Usage
+
+The Horizontal Nav List component takes two props:
+
+- `slug` - which is provided in the props of the page by default
+- `items` - an array of strings for items to render and wrap with a `<Link />` to subheadings
+
+The docs on Gatsbyjs.org use the [gatsby-remark-autolink-headers](/packages/gatsby-remark-autolink-headers/) plugin to automatically apply hover links to heading tags across docs pages. Because it automatically creates links to subheadings on pages like the glossary, the Horizontal Nav List can supply matching links (like `"guide-list"` which would align with the automatically created link at `/docs/docs-and-website-components#guide-list`).
+
+<!-- prettier-ignore -->
+```markdown
+---
+title: Glossary
+---
+
+import HorizontalNavList from "../../www/src/components/horizontal-nav-list.js"
+
+The glossary defines key vocabulary...
+
+---
+
+<HorizontalNavList
+  slug={props.slug}
+  items={["guide-list", "egghead-embed", "pull-quote", "layer-model", "horizontal-navigation-list"]}
+/>
+```
+
+#### Sample
+
+Rendered, it looks like this:
+
+<HorizontalNavList
+items={[
+"guide-list",
+"egghead-embed",
+"pull-quote",
+"layer-model",
+"horizontal-navigation-list",
+]}
+slug={props.slug}
+/>
+
+### Breadcrumb
+
+The `<Breadcrumb />` component is used in layout files to display the hierarchy of pages a user is currently browsing on.
+
+#### Usage
+
+The Breadcrumb component takes one prop:
+
+- `location` - an object provided in the props of page templates by default
+- `itemList` - an object comprised of the docs hierarchical structure
+
+<!-- prettier-ignore -->
+```javascript
+import Breadcrumb from "../../www/src/components/docs-breadcrumb"
+
+<Breadcrumb location={props.location} itemList={itemList} />
+```
+
+_You can also refer to [an example of usage of the Breadcrumb in the Gatsbyjs.org source code](https://github.com/gatsbyjs/gatsby/blob/1d65ce051967dda5c4a89da920fc34692524e237/www/src/templates/template-docs-markdown.js#L82)_
+
+#### Optional `breadcrumbTitle` entries in sidebar files
+
+To alter the title of a doc that is displayed in the Breadcrumb component, a `breadcrumbTitle` is supported as a key in the [sidebar YAML files](https://github.com/gatsbyjs/gatsby/tree/master/www/src/data/sidebars). It is commonly used to provide an abbreviated version of a doc's title when displayed next to its parent page title, e.g. shortening "Adding a Custom webpack Config" to "webpack Config".
+
+#### Sample
+
+Rendered, it looks like this:
+
+<Breadcrumb
+location={{
+    pathname: "/contributing/docs-and-website-components/",
+  }}
+itemList={itemListContributing}
+/>
+
+---
+
+### Table of Contents
+
+The `<TableOfContents />` component is used to render a list of subheaders from a docs page and automatically provide deep links to them.
```

Comment:
should there also be a note how to disable this TOC in the frontmatter in markdown?

Reply:
Yes, I agree -- I thought I commented about the same thing, not sure if the comment went into the ether.